New Financial Institutions. Some minor tweaks.

Zburatorul opened the pull request:
1. I wrote web scrapers for US Bank and Radius Bank. 
2. I figured out the right parameters to pass the finance_dl.ofx module to download transactions from Discover. I added a hack to ofx.py to slow down requests to Discover because they throttle.
3. Modified the Venmo module to ignore lines in the CSV that don't contain transactions (but are actually present in the CSV file produced by Venmo.)
4. Added some sample configs.

Commit:
Venmo CSV files contain a line with some random text and no transaction
data. Ignore this line.

Commit:
USBank downloader. Uses Selenium to log into web portal and download an
OFX file. Works for REI credit card issued by US Bank, not tested on
anything else.

Commit:
Radius Bank downloader. Uses Selenium to log into web portal and download an OFX file.

Commit:
Hack for Discover. Slows down OFX requests.

Commit:
Sample configs for Radius, Discover and US Bank.
Reverted a default date.

Commit:
Sample configs for Radius, Discover and US Bank.

Commit:
Reverted a date value. Merge branch 'master' of github.com:Zburatorul/finance-dl

Commit:
Merge branch 'master' of github.com:Zburatorul/finance-dl

Commit:
Merge branch 'master' of github.com:Zburatorul/finance-dl

Commit:
Fixed to pass Travisc CI tests

jbms commented on finance_dl/radius.py:
```diff
@@ -0,0 +1,274 @@
+    def find_continue_btn(self):
+        for frame in self.for_each_frame():
+            try:
+                return self.driver.find_element_by_xpath('//button[@class="btn bg--aqua"]')
```
If possible you may want to use different criteria --- as this looks like it may be brittle to changes in the website.

Zburatorul replied:
Agree on brittleness. I haven't spent much time staring at HTML in general, so I would appreciate suggestions of the kinds of criteria that are less likely to change.

jbms replied:
Text or attributes that more directly relate to the function rather than the styling tend to be more reliable.  Sometimes you can bypass the HTML altogether and just look at the network requests being made (use network tab of browser developer tools).  If you can just generate the network requests (i.e. the URL and any necessary parameters / post data) directly, that tends to be easiest, fastest and most reliable.  Sometimes that is difficult though, in which case manipulating the UI directly is the way to go..

Zburatorul replied:
Turns out that button check is unnecessary.

jbms commented on finance_dl/radius.py:
```diff
@@ -0,0 +1,274 @@
+    def find_download_button(self):
+        for frame in self.for_each_frame():
+            try:
+                return self.driver.find_element_by_xpath('//i[@class="dropdown icon"]')
```
Same as above, may be brittle.

jbms commented on finance_dl/radius.py:
```diff
@@ -0,0 +1,274 @@
+                 earliest_history_date=None, max_history_days=30,
+                 **kwargs):
+        """
+        @param earliest_history_date: Earliest UTC date for which to retrieve
```
Perhaps just download all available history by default?

Zburatorul replied:
Simplified this by downloading entire year.

jbms commented on finance_dl/radius.py:
```diff
@@ -0,0 +1,274 @@
+"""Retrieves transaction and balance information from RadiusBank.
```
Please rename this module to radiusbank.py for consistency with domain name.

jbms commented:
Thanks for your contribution --- overall looks great, just a few comments.

Commit:
Simplified transaction download login (now downloads year-to-date).
Removed redundant check of "Continue" button presence. Removed code
never used.

Commit:
Added FIs to front page.

jbms commented:
Thanks, looks good.  Any other changes you want to make, or is it ready to merge?

Zburatorul commented:
That's all I got right now.
